app/javascript/src/modules/organisations.js

```diff
@@ -0,0 +1,195 @@
+
+  document
+    .querySelector('#clear-organisation-search')
+    .addEventListener('click', () => {
+      searchInputBox.value = ''
+      hideOrganisationItems()
+      setTimeout(() => { document.querySelector('#screen-reader-messages').innerHTML = 'Search box has been cleared.' }, screenReaderMessageDelay)
+    })
+}
+
+function deselectPreviousOrganisationItem () {
+  const selected = document.querySelector('input:checked')
+  if (selected !== null) { selected.checked = false }
+}
+
+function removeMatchingHtml (obj, regexpsForReplacment) {
+  regexpsForReplacment.forEach(function (regExp) {
+    obj.innerHTML = obj.innerHTML.replace(regExp, '')
+  })
+}
+
+// Find the existing hidden organisation items
+// If they are one of the search matches returned from the V1 api, remove the hidden class
+// and highlight the search terms in the item text
+function showResults (results, inputText) {
+  if (results.length > 0) {
+    deselectPreviousOrganisationItem()
+    const codes = results.map(obj => obj.ccms_opponent_id)
+    let organisationsContainer = document.querySelector('.govuk-radios') // with MP flag on
+    if (organisationsContainer == null) { organisationsContainer = document.querySelector('#organisation-list') } // with MP flag off
+    codes.forEach((code, idx) => {
+      const element = document.getElementById(code)
+
+      // We want to highlight anything in the label or hint text that
+      // matches the user's search criteria
+      const label = element.querySelector('label')
+      const hint = element.querySelector('.govuk-hint')
+      const highlightTagStart = '<mark class="highlight">'
+      const highlightTagEnd = '</mark>'
+      const highlightStartRegExp = RegExp(highlightTagStart, 'gi')
+      const highlightEndRegExp = RegExp(highlightTagEnd, 'gi')
+
+      // Remove existing highlighting
+      removeMatchingHtml(label, [highlightStartRegExp, highlightEndRegExp])
+      removeMatchingHtml(hint, [highlightStartRegExp, highlightEndRegExp])
+
+      // Highlight text that matches the user's input
+      const terms = inputText.split(' ')
+      terms.forEach((term) => {
+        // Negative lookahead conjuction AND negative lookbehind conjuction (non-capturing)
+        // required to exclude any previous search term highlighting which can result in partial HTML
+        // being output to the screen.
+        //
+        // However, this means only the first matched term will be highlighted per object/string/phrase
+        // and this puts it out of sync with the actual fuzzy search matching which works and matches
+        // on a second term even if it is only 1 character.
+        // example: try searching for "ri prison" and only "ri" will be highlighted, despite "prison"
+        // being the type.
+        //
+        // TODO: regex that does not highlight inside a highlight, thereby allowing displaying all matching uniq terms per phrase
```

Comment:
If this TODO is still required can it be linked to a ticket/made clearer? (as I am not clear what would need to be done to resolve).